features/vat-replacement/run.js

```diff
@@ -0,0 +1,213 @@
+    for (const item of items) {
+        let categoryItem = {
+            id : item.id,
+            version : item.version,
+            name : item.name,
+            key : item.key
+        }
+        item.rates.forEach( rate => map.set(rate.id,  categoryItem))
+    }
+    return map
+}
+
+async function printUpdateJson(taxRateDraft, rateIdToCategoryMap ) {
+    const taxCategoryItem = rateIdToCategoryMap.get(taxRateDraft.id)
+
+    const updateJsonObj = {
+        version : taxCategoryItem.version,
+        actions : [
+            {
+                action: "replaceTaxRate",
+                taxRateId: taxRateDraft.id,
+                taxRate: {
+                    name: taxRateDraft.name,
+                    amount: taxRateDraft.amount,
+                    includedInPrice : taxRateDraft.includedInPrice,
+                    country: taxRateDraft.country
+                }
+            }
+        ]
+    }
+    console.log(JSON.stringify(updateJsonObj))
+
+}
+
+async function processTaxRate(ctpClient, validGermanTaxRateList, rateIdToCategoryMap, taxRateType, isDryRun) {
+
+    let taxRateDraftList =  validGermanTaxRateList.filter(rate => rate.amount == taxRateType.original)
+    if (taxRateDraftList.length>=1) {
+        console.log('Current tax rate would be replaced as below : ')
+        for (const taxRateDraft of taxRateDraftList) {
+            taxRateDraft.amount  = taxRateType.new
+            await printUpdateJson(taxRateDraft, rateIdToCategoryMap)
+            if (!isDryRun) {
+                console.log('Start to replace tax rate ... ')
+                await replaceTaxRate(ctpClient, taxRateDraft, rateIdToCategoryMap.get(taxRateDraft.id))
+                console.log('Update finished')
+            }
+        }
+
+    }
+}
+
+async function printPreviewModeWarning() {
+    console.log('********************************************************************')
+    console.log('The script has been run in preview mode. To update the tax rates, ' + '\n' +
+        'you can input argument "-update" in CLI to run this script.')
+    console.log('Please make sure to resolve all tax categories and rates warnings ' + '\n' +
+        'before running the script in update mode.')
+    console.log('********************************************************************')
+}
+(async function main() {
+    const args = process.argv.slice(2);
+    const configOptions = {
+        ctp : {
+            projectKey : process.env.PROJECT_KEY,
+            clientId : process.env.CLIENT_ID,
+            clientSecret : process.env.CLIENT_SECRET
+        },
+        dryRun: true
+    }
+
+    // Check environment variables
+    if (!configOptions.ctp.projectKey || !configOptions.ctp.clientId || !configOptions.ctp.clientSecret) {
+        console.error('Please set project key, client ID and client secret in environment variables')
+        return
+    }
+
+    // Check update mode / preview mode
+    if (args.length>0 && args[0]=='-update') {
```

Comment:
```suggestion
    if (args.length>0 && args[0]==='-update') {
```

In general try to use `===` and `!==` as `==` has a side effect which you may not be aware of and it's harder to check the code in general as I have to think about this side effect: https://developer.mozilla.org/en-US/docs/Glossary/Type_coercion. I see you use in multiple places `==`, replace it with `===`.

Reply:
Replaced.

Reply:
https://github.com/commercetools/commercetools-cli-scripts/pull/1/commits/1d112f0007f22b550e4cbe72d0c0e442a2dc4d82